lib/concurrent/async.rb

```diff
@@ -42,12 +125,35 @@ def validate_argc(obj, method, *args)
         raise ArgumentError.new("wrong number of arguments (#{argc} for #{arity}..*)")
       end
     end
-    module_function :validate_argc
+
+    # @!visibility private
+    def self.included(base)
+      base.extend(ClassMethods)
+      base.send(:private_class_method, :new)
+      super(base)
+    end
+
+    # @!visibility private
+    def self.extended(base)
+      base.extend(ClassMethods)
+      base.send(:private_class_method, :new)
+      super(base)
+    end
+
+    # @!visibility private
+    module ClassMethods
+      def create(*args, &block)
+        obj = self.send(:new, *args, &block)
+        obj.send(:init_synchronization)
```

Comment:
I know it breaks the API, but I think that overriding `new` is a questionable practice. Using a factory method makes it clear that the object creation process has been modified. I'll add some deprecation warnings to and backward-compatable methods.